### PR TITLE
feat: guix to replace gitian and its dashification

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
This PR switches building dash from gitian to Guix containers. [Guix](https://www.gnu.org/software/guix/manual/en/html_node/Features.html) is a transactional package manager much like Nix, but unlike Nix, it has more of a focus on [bootstrappability](https://www.gnu.org/software/guix/manual/en/html_node/Bootstrapping.html) and [reproducibility](https://www.gnu.org/software/guix/blog/tags/reproducible-builds/) which are attractive for security-sensitive projects like dash, bitcoin.

Prior work: https://github.com/dashpay/dash/pull/5194 and https://github.com/dashpay/dash/pull/5237
Related issue: https://github.com/dashpay/dash-issues/issues/47

## What was done?
 - backports that have docs update for releasing process (replacement gitian to guix)
   - bitcoin/bitcoin#21711
   - bitcoin/bitcoin#22511
   - bitcoin/bitcoin#22642
   - bitcoin/bitcoin#22654
 - dashification of all guix related components: docs, scripts, etc that were postponed till now.
 - added Dockerfile for guix build (authored by @PastaPastaPasta). 

Things to do before merging PR:
 - test signing binaries, release process, gitian builds
 - update [seeds TODO](/contrib/seeds/README.md). 
 - create guix.sigs repo

## How Has This Been Tested?
Followed instruction [guix/README.md](contrib/guix/README.md) and got guix builds.
**To test: signing binaries (not tested yet).** Related instruction: [Release Process](doc/release-process.md)
**To test: gitian build.** Related instruction: [Release Process](doc/release-process.md)

Hashes to compare:
```sh
$  cat guix-build-5dd78dfd9629/output/*/SHA256SUMS* | LOCALE=C sort -k2 
d1ae811b9d0ff8ff58ad9c06e70e37f8219d7ef907b6d55d8b9c93e43b22d94d  aarch64-linux-gnu/dashcore-5dd78dfd9629-aarch64-linux-gnu-debug.tar.gz
588192d63c76e9c60aee6c2a100fad7342c06bf9991fc400c1828b36365e8ba7  aarch64-linux-gnu/dashcore-5dd78dfd9629-aarch64-linux-gnu.tar.gz
26dea95600701cee10851a8c097a1482350651a67d6ef477abd46af8ff1bcb47  arm-linux-gnueabihf/dashcore-5dd78dfd9629-arm-linux-gnueabihf-debug.tar.gz
0f0290c87fffc6ddf88eaf42e36ea89c5851020a340a9e03e3998878cc326e38  arm-linux-gnueabihf/dashcore-5dd78dfd9629-arm-linux-gnueabihf.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
72c07a447c2ff61557f88f8508a0f332c3ddb7648d2266b3b1ce3e275bb286c1  riscv64-linux-gnu/dashcore-5dd78dfd9629-riscv64-linux-gnu-debug.tar.gz
1ffb5a64b6b11ac90427905db437fb0436518a633fb6a2eb73388f252b6bfd2c  riscv64-linux-gnu/dashcore-5dd78dfd9629-riscv64-linux-gnu.tar.gz
bb9305c39a928513bd444c6b9e8998d8cf136b64cc27cca96dc3b79e78ab5c07  x86_64-apple-darwin18/dashcore-5dd78dfd9629-osx64.tar.gz
4f3bebc12057de597fa16df84031977dcfc94140cb4a7dbf04c22b188eedbc7f  x86_64-apple-darwin18/dashcore-5dd78dfd9629-osx-unsigned.dmg
f700ee26b7a99a7f017d8c91963989e7d3260611ce589c44547810a47470e23b  x86_64-apple-darwin18/dashcore-5dd78dfd9629-osx-unsigned.tar.gz
dbda476f50ebade9a163f0276189dad7f0f8fc5755a603c7f4162a03f5e8ec44  x86_64-linux-gnu/dashcore-5dd78dfd9629-x86_64-linux-gnu-debug.tar.gz
7fe233ffad7ae92eafa8a5c4180781a17957cb49d39d0f0521ec4c307bb085ad  x86_64-linux-gnu/dashcore-5dd78dfd9629-x86_64-linux-gnu.tar.gz
10ce8bd9fa28a8ceebf42474ba91d7d9a78956dbe24ae452e1e209bf95b4525e  x86_64-w64-mingw32/dashcore-5dd78dfd9629-win64-debug.zip
42693fd70feb91c79929177309345b87de4b5309e236519bf6963fe509d0d546  x86_64-w64-mingw32/dashcore-5dd78dfd9629-win64-setup-unsigned.exe
112aadb4b9c166f9a6740800c68af58aee94c69af87b9522eaeb2aac1530e3a7  x86_64-w64-mingw32/dashcore-5dd78dfd9629-win64.zip
eac891cdea9d508f28a4720491cc8deb4f5eed1b9eb767c04ab82a131e8111e7  x86_64-w64-mingw32/dashcore-5dd78dfd9629-win-unsigned.tar.gz
```

## Breaking Changes
That's not really breaking changes, gitian should still work. 
But the changes are major and should be included in release notes at least.

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
My results
```
pasta@pasta1:~/dash$ cat guix-build-5dd78dfd9629/output/*/SHA256SUMS* | LOCALE=C sort -k2
d1ae811b9d0ff8ff58ad9c06e70e37f8219d7ef907b6d55d8b9c93e43b22d94d  aarch64-linux-gnu/dashcore-5dd78dfd9629-aarch64-linux-gnu-debug.tar.gz
588192d63c76e9c60aee6c2a100fad7342c06bf9991fc400c1828b36365e8ba7  aarch64-linux-gnu/dashcore-5dd78dfd9629-aarch64-linux-gnu.tar.gz
26dea95600701cee10851a8c097a1482350651a67d6ef477abd46af8ff1bcb47  arm-linux-gnueabihf/dashcore-5dd78dfd9629-arm-linux-gnueabihf-debug.tar.gz
0f0290c87fffc6ddf88eaf42e36ea89c5851020a340a9e03e3998878cc326e38  arm-linux-gnueabihf/dashcore-5dd78dfd9629-arm-linux-gnueabihf.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
7b48c5f760a530f4f7d5ada5c140cf0f53881d297735d0553d5e373919c4d25d  dist-archive/dashcore-5dd78dfd9629.tar.gz
72c07a447c2ff61557f88f8508a0f332c3ddb7648d2266b3b1ce3e275bb286c1  riscv64-linux-gnu/dashcore-5dd78dfd9629-riscv64-linux-gnu-debug.tar.gz
1ffb5a64b6b11ac90427905db437fb0436518a633fb6a2eb73388f252b6bfd2c  riscv64-linux-gnu/dashcore-5dd78dfd9629-riscv64-linux-gnu.tar.gz
4f3bebc12057de597fa16df84031977dcfc94140cb4a7dbf04c22b188eedbc7f  x86_64-apple-darwin18/dashcore-5dd78dfd9629-osx-unsigned.dmg
f700ee26b7a99a7f017d8c91963989e7d3260611ce589c44547810a47470e23b  x86_64-apple-darwin18/dashcore-5dd78dfd9629-osx-unsigned.tar.gz
bb9305c39a928513bd444c6b9e8998d8cf136b64cc27cca96dc3b79e78ab5c07  x86_64-apple-darwin18/dashcore-5dd78dfd9629-osx64.tar.gz
dbda476f50ebade9a163f0276189dad7f0f8fc5755a603c7f4162a03f5e8ec44  x86_64-linux-gnu/dashcore-5dd78dfd9629-x86_64-linux-gnu-debug.tar.gz
7fe233ffad7ae92eafa8a5c4180781a17957cb49d39d0f0521ec4c307bb085ad  x86_64-linux-gnu/dashcore-5dd78dfd9629-x86_64-linux-gnu.tar.gz
eac891cdea9d508f28a4720491cc8deb4f5eed1b9eb767c04ab82a131e8111e7  x86_64-w64-mingw32/dashcore-5dd78dfd9629-win-unsigned.tar.gz
10ce8bd9fa28a8ceebf42474ba91d7d9a78956dbe24ae452e1e209bf95b4525e  x86_64-w64-mingw32/dashcore-5dd78dfd9629-win64-debug.zip
42693fd70feb91c79929177309345b87de4b5309e236519bf6963fe509d0d546  x86_64-w64-mingw32/dashcore-5dd78dfd9629-win64-setup-unsigned.exe
112aadb4b9c166f9a6740800c68af58aee94c69af87b9522eaeb2aac1530e3a7  x86_64-w64-mingw32/dashcore-5dd78dfd9629-win64.zip
```

**Comment by knst** [comment]
I can't sign binaries successfully by current implementation.
I will prepare new PR that has related backports and will fix signing process after.

File: doc/release-process.md [hunk 1]
```diff
@@ -24,18 +24,17 @@ Before every major release:
...
 
-If you're using the automated script (found in [contrib/gitian-build.py](/contrib/gitian-build.py)), then at this point you should run it with the "--setup" command. Otherwise ignore this.
+Install Guix using one of the installation methods detailed in
+[contrib/guix/INSTALL.md](/contrib/guix/INSTALL.md).
 
 Check out the source code in the following directory hierarchy.
 
 	cd /path/to/your/toplevel/build
-	git clone https://github.com/dashpay/gitian.sigs.git
+	git clone https://github.com/dashpay/guix.sigs.git
```
[the remark this answers is not in the document]
Comment (knst) [reply doc/release-process.md:36]: the hashes of artefacts of gitian and guix are not the same.
I think better to have new repo to avoid a confusion

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit; although I don't love significantly changing the release process doc before we have signed binaries and everything figured out, but gitian still exists after this PR, so it's probably all fine

**Review by UdjinM6 — APPROVED** [review]
LGTM, ACK (same hashes for 5dd78dfd9629)

**Pull request merged**